web/web.go

```diff
@@ -300,7 +300,13 @@ func New(logger log.Logger, o *Options) *Handler {
 	factoryTr := func(_ context.Context) api_v1.TargetRetriever {
 		return h.scrapeManager
 	}
-	h.apiV1 = api_v1.NewAPI(h.queryEngine, h.storage, factoryTr, h.notifier,
+	factoryAr := func(_ context.Context) api_v1.AlertmanagerRetriever {
+		return h.notifier
+	}
+	FactoryRr := func(_ context.Context) api_v1.RulesRetriever {
```

Comment:
All of this should be just inlined, especially if you can have just one line (:  WDYT? No need for this explicitness.

Reply:
+ less LOC if you do so

Reply:
Happy to, done.